### PR TITLE
Expose ip address to .sub. templates

### PR DESCRIPTION
**Pull request by jgraham**
This doesn't enable anything really new, but does make it a little easier
to write tests for features that have different behaviour depending on the
use of host name vs ip address for the server

### CODE REVIEW

File: tools/wptserve/wptserve/pipes.py [hunk 1]
```diff
@@ -347,6 +347,8 @@ def sub(request, response, escape_type="html"):
       'server, 'scheme', 'host', 'hostname', 'port', 'path' and 'query'.
       'server' is scheme://host:port, 'host' is hostname:port, and query
       includes the leading '?', but other delimiters are omitted.
+    ip address
```
Comment (Hexcles) [inline tools/wptserve/wptserve/pipes.py]:
```suggestion
    ip_address
```

File: tools/wptserve/wptserve/pipes.py [hunk 2]
```diff
@@ -484,6 +488,8 @@ def config_replacement(match):
             value = request.server.config.all_domains[""]
         elif field == "host":
             value = request.server.config["browser_host"]
+        elif field == "ip_address":
+            value = request.server.config["server_host"]
```
Comment (Hexcles) [inline tools/wptserve/wptserve/pipes.py]: Question: do we guarantee that `server_host` is an IP? Can it be e.g. `localhost`? Even if the default is an IP, some people might overwrite it to `localhost` in their custom config without noticing (as everything else works regardless I assume).

Comment (Hexcles) [reply tools/wptserve/wptserve/pipes.py]: ?

**Comment by jgraham** [comment]
Right, I forgot to comment and basically say "you're right; this doesn't work as implemented". I still think this should be possible but it might need a bit more work. I'll close this for now to get it off the radar.

**Pull request closed without merging**